include/NovelRT/Maths/GeoBounds.h

```diff
@@ -10,25 +10,85 @@
 
 namespace NovelRT::Maths
 {
+    /**
+     * @brief Represents a bounding box in 2D space with rotational support.
+     */
     class GeoBounds
     {
     public:
+        /// @brief The centre of the GeoBounds instance as a position in X,Y coordinates in the space used during its creation.
         GeoVector2F position;
+        /// @brief The size represented by Width and Height. This is subject to the space used during the creation of the GeoBounds instance.
         GeoVector2F size;
+        /// @brief The rotation of the GeoBounds instance represented in Degrees.
         float rotation;
 
+        /**
+         * @brief Instantiates a GeoBounds instance with the specified position, size and rotation.
+         *
+         * @param position A starting point represented by X and Y in a given space.
+         * @param size A size represented by Width and Height in a given space.
+         * @param rotation A starting rotation for the GeoBounds instance that is represented in degrees.
+         * @return Newly constructed GeoBounds object containing the position, size, and rotation components.
+         */
         GeoBounds(GeoVector2F position, GeoVector2F size, float rotation) noexcept;
+
+        /**
+         * @brief Determines if a given GeoVector2F is within the confines of this bounding box.
+         * @param point The GeoVector2F to be evaluated.
+         * @return true if the given GeoVector2F is within the confines of this GeoBounds instance, false otherwise.
+         */
         [[nodiscard]] bool pointIsWithinBounds(GeoVector2F point) const noexcept;
+
+        /**
+         * @brief Evaluates whether or not if another GeoBounds instance intersects with this GeoBounds. 
+         *
+         * This implementation does not support rotation, so only AABB intersection computation is allowed.
+         *
+         * @param otherBounds the other GeoBounds instance to evaluate against this GeoBounds instance.
+         * @exception Exceptions::NotSupportedException If either bounding box has a rotation set to a value other than 0.
+         * @return true if the otherBounds intersects with this bounding box, otherwise false.
```

Comment:
```suggestion
         * @return true if the otherBounds intersects with this bounding box, otherwise false.
         * @exception Exceptions::NotSupportedException If either bounding box has a rotation set to a value other than 0.
```

Reply:
Applied this suggestion in https://github.com/novelrt/NovelRT/pull/483/commits/1a44662d0f61647b5a68f916198914a6302c464e